src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -223,64 +231,193 @@ private UUID getParticipantEntitySetId( UUID studyId ) {
 
     }
 
-    private UUID getParticipantEntityKeyId (String participantId, UUID studyId) {
-        if (studyParticipants.containsKey(studyId)) {
-            Map<String, UUID> participantIdToEKMap = studyParticipants.get(studyId);
+    private UUID getParticipantEntityKeyId( String participantId, UUID studyId ) {
+        if ( studyParticipants.containsKey( studyId ) ) {
+            Map<String, UUID> participantIdToEKMap = studyParticipants.get( studyId );
 
-            if (participantIdToEKMap.containsKey(participantId)) {
-                return participantIdToEKMap.get(participantId);
+            if ( participantIdToEKMap.containsKey( participantId ) ) {
+                return participantIdToEKMap.get( participantId );
             } else {
-                logger.error("Unable to get participantEntityKeyId. participant {} not associated with studId {} ", participantId, studyId);
+                logger.error( "Unable to get participantEntityKeyId. participant {} not associated with studId {} ",
+                        participantId,
+                        studyId );
             }
         } else {
-            logger.error("Unable to get participantEntityKeyId of participantId {}. StudyId {}  not found ", participantId, studyId);
+            logger.error( "Unable to get participantEntityKeyId of participantId {}. StudyId {}  not found ",
+                    participantId,
+                    studyId );
         }
 
         return null;
     }
 
-    //  TODO: add in throws exception!
-    @Override
-    public Integer logData(
-            UUID studyId,
+    // return an OffsetDateTime with time 00:00
+    private String parseDateTime( String dateTime ) {
+        return OffsetDateTime
+                .parse( dateTime )
+                .withHour( 0 )
+                .withMinute( 0 )
+                .withSecond( 0 )
+                .withNano( 0 )
+                .toString();
+    }
```

Comment:
The method name does not accurately reflect the actual implementation. It's not a parser.